src/pool/srv_pool.c

```diff
@@ -1694,10 +1696,12 @@ pool_svc_map_dist_cb(struct ds_rsvc *rsvc)
 	}
 
 	rc = ds_pool_iv_map_update(svc->ps_pool, map_buf, map_version);
-	if (rc != 0)
+	if (rc != 0) {
 		D_ERROR(DF_UUID": failed to distribute pool map %u: %d\n",
 			DP_UUID(svc->ps_uuid), map_version, rc);
-
+		D_GOTO(out, rc);
+	}
+	svc->ps_global_map_version = max(svc->ps_global_map_version, map_version);
```

Comment:
If (svc->ps_global_map_version > map_version), do we still need above ds_pool_iv_map_update() with old map_version? or re-fetch the map_version under such case?

Reply:
Not sure I follow the question, but this function will be called on the leader only,  to distribute the pool map to other targets as requested, so when the pool map is updated, the global map version is updated.

> we still need above ds_pool_iv_map_update() with old map_version?

since this is the leader, I assume this will never happen? @liw  or I miss your question? @Nasf-Fan